lib/trino-record-decoder/src/main/java/io/trino/decoder/avro/AvroColumnDecoder.java

```diff
@@ -275,7 +281,7 @@ private static void serializePrimitive(BlockBuilder blockBuilder, Object value,
             return;
         }
 
-        if ((value instanceof Integer || value instanceof Long) && (type instanceof BigintType || type instanceof IntegerType || type instanceof SmallintType || type instanceof TinyintType)) {
+        if ((value instanceof Integer || value instanceof Long) && (type instanceof BigintType || type instanceof IntegerType || type instanceof SmallintType || type instanceof TinyintType || type instanceof DateType || type instanceof TimeType || type instanceof TimestampType)) {
```

Comment:
`TimeType` and `TimestampType` contain other precisions in addition to `TIME_MILLIS` & `TIMESTAMP_MILLIS`. 
 Also the internal implementation of `LongTimestampType` & `ShortTimestampType` extending `TimestampType` differ. Why do you want to handle all precisions instead of `TIME_MILLIS` & `TIMESTAMP_MILLIS`?